### ISSUE TITLE
defaultTimeout isn't applied to new-syntax steps

### ISSUE THREAD

**Issue by tshedor** [jonsamwell/flutter_gherkin#81]
Ay, sorry to open this as an issue instead of a PR but I couldn't find where the context/configuration was being built. 

When the new steps from #70 were converted into [new syntax steps](https://github.com/jonsamwell/flutter_gherkin/commit/b2f86e04464f54f77dfa9d830c5b1b5c9b8e1c5a#diff-3cce40c99a14a7e4ff55d8a1110e8105R17) they lost the `defaultTimeout` applied from `TestConfiguration`. I thought this would be accessible via `context.configuration.timeout` but `context.configuration == null`, i.e.

```dart
await FlutterDriverUtils.waitUntil(
  context.world.driver,
  () {
    return existence == Existence.absent
        ? FlutterDriverUtils.isAbsent(
            context.world.driver,
            find.byValueKey(keyString),
          )
        : FlutterDriverUtils.isPresent(
            context.world.driver,
            find.byValueKey(keyString),
          );
  },
  timeout: context.configuration.timeout
);
```

Any idea how to generate a `StepConfiguration` that uses `TestConfiguration#defaultTimeout` and apply it to context for each step?

**Comment by jonsamwell** [comment]
Hi @tshedor 

Good spot, it looks like it the passed in configuration is null or the timeout is null it should be populated from the defaults passed to the step from the runner https://github.com/jonsamwell/dart_gherkin/blob/master/lib/src/gherkin/steps/step_definition_implementations.dart#L188.

I am away at the moment but can sort this at the weekend.

**Comment by tshedor** [comment]
@jonsamwell thanks for the quick response, I'll take a look tomorrow. 

From a higher level, is there a performance benefit to the function-style steps? I personally like the original class-based definitions because they allow using `extends` to add build more complex steps in an iterative way (i.e. chaining `super.executeStep()`). Class-based also feels more future-proof: if the protocol of `WhenWithWorld` was ever changed, debugging outdated properties on a class is cleaner to debug than through a function. And I know the functions remove some boilerplate, but I believe at the higher cost of legibility.

**Comment by jonsamwell** [comment]
Fixed in release 1.1.8+9
